### DISCUSSION TITLE
Ingest only 1 column / computed column values

### DISCUSSION THREAD

**Discussion by gagangoku** [ClickHouse/ClickHouse#57028] [Q&A]
I want to use Clickhouse for some custom logic (really a language) I have built, which includes data transformations. Here's an example of joining two tables on some conditions:

```
// Load team: userPhone, role
tbl.Team = LOAD_CSV('gs://my-bucket/team.csv')

// Load messages: userId, groupId, body, timestamp, type
tbl.Messages = LOAD_CSV('gs://my-bucket/messages.csv')

// Add a role column to messages (lookup from Team table where userPhone and userId match)
tbl.Messages.role = FIRST ( LOOKUP ( tbl.Team.role , tbl.Team.userPhone == tbl.Messages.userId ) )
```

I am generating the data in my custom logic outside, and am planning to ingest the data produced into Clickhouse in one shot. I see that there's a way to ingest data in a columnar format, which is great.

But is it possible to insert a column and only ingest the values for that column ? Imagine if the user wants to add one more computed column to the table, but doesn't have to reingest everything.

**Comment by den-crane** [comment]
Upsert? No such feature.

**Comment by gagangoku** [comment]
https://clickhouse.com/blog/handling-updates-and-deletes-in-clickhouse#updating-and-deleting-using-joins seems to be the closest to what i'm looking for
